tests/Traits/UserAuthenticatorTrait.php

```diff
@@ -23,26 +23,26 @@
 namespace OAT\SimpleRoster\Tests\Traits;
 
 use OAT\SimpleRoster\Entity\User;
+use OAT\SimpleRoster\Service\JWT\JWTManager;
 use Symfony\Bundle\FrameworkBundle\KernelBrowser;
-use Symfony\Component\BrowserKit\Cookie;
-use Symfony\Component\HttpFoundation\Session\Session;
-use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
 
 trait UserAuthenticatorTrait
 {
-    protected function logInAs(User $user, KernelBrowser $kernelBrowser): void
+    protected function logInAs(User $user, KernelBrowser $kernelBrowser): string
     {
-        /** @var Session $session */
-        $session = $kernelBrowser->getContainer()->get('session');
+        $ttl = static::$container->getParameter('app.jwt.access_token_ttl');
+        $refreshTtl = static::$container->getParameter('app.jwt.refresh_token_ttl');
 
-        // the firewall context defaults to the firewall name
-        $firewallContext = 'api';
+        /** @var JWTManager $jwt */
+        $jwt = static::$container->get(JWTManager::class);
 
-        $token = new UsernamePasswordToken($user, null, $firewallContext, $user->getRoles());
-        $session->set('_security_' . $firewallContext, serialize($token));
-        $session->save();
+        $accessToken = $jwt->create($user, $ttl);
 
-        $cookie = new Cookie($session->getName(), $session->getId());
-        $kernelBrowser->getCookieJar()->set($cookie);
+        $refreshToken = $jwt->create($user, $refreshTtl);
+        $jwt->storeTokenInCache($refreshToken, $refreshTtl);
+
+        $kernelBrowser->setServerParameter('HTTP_Authorization', sprintf('Bearer %s', $accessToken));
+
+        return (string)$refreshToken;
```

Comment:
Whaaaat..? What you should do here is to actually call the login endpoint with the user credentials, grab the token from the response and return it at the end so it can be used for further API calls. Please refactor

Reply:
Good point. But, it also means (as I wrote to you) that we would be missing possibility to check logs via TestHandler in that case. So, I just did it as it was for normal auth flow -> without sending real request, just creating tokens and setting access one into server params.
Now the question: are you fully sure that it would be better to use "normal" flow with requesting token via API *_for tests_*, instead of just setting it?

Reply:
Yes

Reply:
You don't care about the login logs 99% of the time, the only place you want to test it when you test the login endpoint itself. In other test cases you are not asserting on it

Reply:
> You don't care about the login logs 99% of the time, the only place you want to test it when you test the login endpoint itself. In other test cases you are not asserting on it
Not login logs. What I meant is - if you remember on prev week we've discussed the problem I had with one test failing, because TestHandler didn't see any log records. As I've found - this is because sending the request to login, as after the request the logger handlers would be cleared. So, yeah, why not I can switch it back to use request for authorization via JWT, BUT, this would mean that I would drop some test assertions checking the log records, as it would be impossible.
So, are we agreed on that?

Reply:
No we don't. If you authenticate the user with this trait AND set up the test log handler after, you'll be able to assert on logs. Let's refactor, and then I can help you with the log assertions. Here is an example by the way where we assert on logs after authentication: `GetUserAssignmentLtiLinkActionTest->testItLogsSuccessfulLtiRequestCreation()`